js_modules/dagit/packages/core/src/runs/LogsProvider.tsx

```diff
@@ -210,44 +232,38 @@ const LogsProviderWithQuery = (props: LogsProviderWithQueryProps) => {
         // We have to stop polling in order to update the `after` value.
         stopPolling();
 
-        const slice = () => {
-          const count = nodes.length;
-          if (data?.pipelineRunOrError.__typename === 'Run') {
-            return data?.pipelineRunOrError.eventConnection.events.map((event, ii) => ({
-              ...event,
-              clientsideKey: `csk${count + ii}`,
-            }));
-          }
-          return [];
-        };
-
-        const newSlice = slice();
-        setNodes((current) => [...current, ...newSlice]);
-        if (data?.pipelineRunOrError.__typename === 'Run') {
-          setCursor(data.pipelineRunOrError.eventConnection.cursor);
+        if (data?.pipelineRunOrError.__typename !== 'Run') {
+          return;
         }
 
-        const status =
-          data?.pipelineRunOrError.__typename === 'Run' ? data?.pipelineRunOrError.status : null;
+        const run = data.pipelineRunOrError;
+        const queued = run.eventConnection.events;
```

Comment:
FYI, I want to land https://github.com/dagster-io/dagster/pull/8077 which will pull the events off of the run object and into its own top-level query object